Dockerfile

```diff
@@ -68,27 +66,39 @@ RUN apt-get update \
   && rm -rf /var/lib/apt/lists/*
 
 ARG SPARK_VERSION
-ARG SPARK_VARIANT
+ARG HADOOP_VERSION
 ARG SPARK_PY4J
 
+ENV HADOOP_HOME /opt/hadoop
+ENV PATH ${PATH}:${HADOOP_HOME}/bin:${HADOOP_HOME}/sbin
+ENV HADOOP_CONF_DIR ${HADOOP_HOME}/etc/hadoop
+ENV HDFS_CONF_DIR ${HADOOP_CONF_DIR}
+ENV YARN_CONF_DIR ${HADOOP_CONF_DIR}
+ENV HIVE_CONF_DIR ${HADOOP_CONF_DIR}
+
 ENV SPARK_HOME=/opt/spark-${SPARK_VERSION}
 ENV PATH=$PATH:${SPARK_HOME}/bin
 ENV PYTHONPATH=${SPARK_HOME}/${SPARK_PY4J}:${SPARK_HOME}/python
 ENV PYSPARK_SUBMIT_ARGS="--driver-memory 8g --py-files ${SPARK_HOME}/python/lib/pyspark.zip pyspark-shell"
 
 # Download Spark
 ARG SPARK_EXTRACT_LOC=/sparkbin
-RUN set -eux && \
+RUN set -eux && \\
+    curl http://www-us.apache.org/dist/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz | \
+    tar -xz -C /opt/ && \
+    mv /opt/hadoop-${HADOOP_VERSION} /opt/hadoop && \
     mkdir -p ${SPARK_EXTRACT_LOC} && \
-    curl https://www.mirrorservice.org/sites/ftp.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-${SPARK_VARIANT}.tgz |\
+    curl https://www.mirrorservice.org/sites/ftp.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION:0:3}.tgz | \
     tar -xz -C ${SPARK_EXTRACT_LOC} && \
     mkdir -p ${SPARK_HOME} && \
-    mv ${SPARK_EXTRACT_LOC}/spark-${SPARK_VERSION}-bin-${SPARK_VARIANT}/* ${SPARK_HOME} && \
+    mv ${SPARK_EXTRACT_LOC}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION:0:3}/* ${SPARK_HOME} && \
     rm -rf ${SPARK_EXTRACT_LOC} && \
     echo SPARK_HOME is ${SPARK_HOME} && \
     ls -al --g ${SPARK_HOME}
 
 # Less verbose logging
 COPY log4j.properties.production ${SPARK_HOME}/conf/log4j.properties
 
-
+## To build your own image:
+ONBUILD COPY hadoop/conf/ ${HADOOP_CONF_DIR}/
```

Comment:
I don't know... We could be targeting three or four different systems, I don't see the point of copying in the config files

Reply:
True that. Will remove this and update the Readme.